pkg/crdpuller/discovery.go

```diff
@@ -413,6 +413,12 @@ var knownPackages map[string]map[string]apiextensionsv1.JSONSchemaProps = map[st
 		// generate validation for it.
 		"ObjectMeta": apiextensionsv1.JSONSchemaProps{
 			Type: "object",
+
+			// CHANGE vs the OperatorSDK code:
+			// Add `x-preserve-unknown-fields: true` here since it is necessary when metadata is not at the top-level
+			// (for example in Deployment spec.template.metadata), in order not to prune unknown fields and avoid
+			// emptying metadata
+			XPreserveUnknownFields: boolPtr(true),
```

Comment:
I'm really confused why ObjectMeta wouldn't be a struct like any other.  This is fine for now but we're actually going to have to *know* which metadata fields different objects have (they won't all have the same) and so CRD normalization has to have a concrete list.

Reply:
I partly agree with you.
However currently metadata has quite complex context like the `managedFields` field that obviously are not really designed to be structural-schema-compliant: the CRD generated from this isn't compliant on the server-side.
I assume this might be what explains the choice made by the controller-tools team in this regard.
But sure, in the future we could decide to build the metadata schema ourselves for metadata that are not at the top-level.
FTR metadata at the top-level are expected to be an empty object Json Schema definition by the CRD validator on the apiServer side.